NEXT-31249 - async low priority queue documentation (#1153)
shopware/docs: guides/hosting/infrastructure/message-queue.md

## Overview

::: warning

> This guide contains the `async_low_priority` queue which is only available in version 6.5.7.0 and above. Configuring the messenger to consume this queue will fail, if it does not exist.

Reviewer: ```suggestion
Parts of this guide refer to the `async_low_priority` queue which is only available in version 6.5.7.0 and above. Configuring the messenger to consume this queue will fail, if it does not exist.
```

> This guide contains the `async_low_priority` queue which is only available in version 6.5.7.0 and above. You must not configure this queue in older versions as the messenger:consume command will fail.

Reviewer: ```suggestion
This guide contains the `async_low_priority` queue which is only available in version 6.5.7.0 and above. Configuring the messenger to consume this queue will fail, if it does not exist.
```

bin/console messenger:consume async --time-limit=60 --memory-limit=128M
```

> You can also configure the command to consume messages from multiple transports as it is recommended:

Reviewer 2: also link to the symfony docs on multiple queues?

Author: I will add the same documentation regarding the prioritized transports here :+1:

bin/console messenger:consume async --time-limit=60 --memory-limit=128M
```

You can also configure the command to consume messages from multiple transports, in order to prioritize them to your needs, as it is recommended by the [Symfony documentation](https://symfony.com/doc/current/messenger.html#prioritized-transports):

```bash

```bash

> bin/console messenger:consume async async_low_priority

Reviewer: ```suggestion
bin/console messenger:consume async failed async_low_priority
```

Restart=always
# Change the path to your shop path
WorkingDirectory=/var/www/html

> ExecStart=php /var/www/html/bin/console messenger:consume --time-limit=60 --memory-limit=512M async async_low_priority

Reviewer 2: do we need to mention that this is only there from 6.5.x? Can you test whant happens when you start a worker for a queue that does not exist?

Maybe we need to split this MR for what already be done in 6.5 and what needs to change in 6.6 (e.g. how the command is executed)

Author: I will add a warning that you must not add this in earlier versions as the command will fail :+1:

### Admin worker

::: warning

> The `transports` option can only be configured with the `low_priority` transport if you are on version 6.5.7.0 or above. You must not add the `low_priority` transport in lower versions as the admin worker will fail when it tries to consume a non-existent transport.

Reviewer 3: ```suggestion
The `transports` option can only be configured with the `low_priority` transport if you are on version 6.5.7.0 or above. You can't add the `low_priority` transport in lower versions, as the admin worker will fail when it tries to consume a non-existent transport.
```

    admin_worker:
        enable_admin_worker: true
        poll_interval: 30

>         transports: ["async", "async_low_priority"]

Reviewer: ```suggestion
        transports: ["async", "failed", "async_low_priority"]
```

@keulinho was it just missing or is the admin worker not supposed to consume the `failed` transport? 🤔

Author: I wanted to ask the same for the command. In the symfony docs they recommend to run the failed queue manually and have a look at the failed messages :sweat_smile:

Reviewer 2: Yes so it was intentional ;)
